tests/test_nonblocking_stream.py

```diff
@@ -70,9 +84,21 @@ def test_send_event(self):
         assert_equal(mock_sent_meta_data, meta_data)
         assert_equal(mock_sent_message_data, message_data)
 
+    def test_send_hard_to_encode_data(self):
+        s = nonblocking_stream.NonblockingStream('test_stream', 'pkey')
+        test_data = generate_messy_test_data()
+        s.put(**test_data)
+        meta_data, message_data = generate_transmitted_record(test_data)
+        mock_sent_meta_data, mock_sent_message_data = (
+            nonblocking_stream.threadLocal
+            .zmq_socket.send_multipart.calls[0][0][0])
+        assert_equal(mock_sent_meta_data, meta_data)
```

Comment:
yeah, we should check though they are encoded to the right value. like seconds vs isostring right?